src/util/write-payload.ts

```diff
@@ -33,9 +33,9 @@ export class WritePayload<T extends JSORMBase> {
     const attrs: ModelRecord<T> = {}
 
     this._eachAttribute((key, value) => {
-      const snakeKey = underscore(key)
+      const snakeKey: keyof ModelRecord<T> = underscore(key) as any
 
-      if (!this.model.isPersisted || this.model.changes()[key]) {
+      if (!this.model.isPersisted || (<any>this.model.changes())[key]) {
```

Comment:
`any`?

Reply:
Something about typescript 2.7 doesn't like indexing into the `ModelRecord<T>` type we created, so need to cast to get it to behave.